apps/wallet/src/ui/app/components/transactions-card/Transaction.tsx

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    getExecutionStatusType,
+    getTransactionKindName,
+    getMoveCallTransaction,
+    getExecutionStatusError,
+    getTransferObjectTransaction,
+    SUI_TYPE_ARG,
+} from '@mysten/sui.js';
+import { useMemo } from 'react';
+import { Link } from 'react-router-dom';
+
+import { TxnTypeLabel } from './TxnActionLabel';
+import { TxnIcon } from './TxnIcon';
+import { TxnImage } from './TxnImage';
+import { CoinBalance } from '_app/shared/coin-balance';
+import { DateCard } from '_app/shared/date-card';
+import { Text } from '_app/shared/text';
+import { getEventsSummary, getAmount, notEmpty } from '_helpers';
+import { useGetTxnRecipientAddress } from '_hooks';
+
+import type {
+    SuiTransactionResponse,
+    SuiAddress,
+    TransactionEffects,
+    SuiEvent,
+} from '@mysten/sui.js';
+
+const getTxnEffectsEventID = (
+    txEffects: TransactionEffects,
+    address: string
+): string[] => {
+    const events = txEffects?.events || [];
+    const objectIDs = events
+        ?.map((event: SuiEvent) => {
+            const data = Object.values(event).find(
+                (itm) => itm?.recipient?.AddressOwner === address
+            );
+            return data?.objectId;
+        })
+        .filter(notEmpty);
+    return objectIDs;
+};
+
+export function Transaction({
+    txn,
+    address,
+}: {
+    txn: SuiTransactionResponse;
+    address: SuiAddress;
+}) {
+    const { certificate } = txn;
+    const executionStatus = getExecutionStatusType(txn);
+    const txnKind = getTransactionKindName(certificate.data.transactions[0]);
+    const { coins: eventsSummary } = getEventsSummary(txn.effects, address);
+
+    const objectId = useMemo(() => {
+        const transferId = getTransferObjectTransaction(
+            certificate.data.transactions[0]
+        )?.objectRef?.objectId;
+        return transferId
+            ? transferId
+            : getTxnEffectsEventID(txn.effects, address)[0];
+    }, [address, certificate.data.transactions, txn.effects]);
+
+    const amountByRecipient = getAmount(
+        certificate.data.transactions[0],
+        txn.effects
+    );
+
+    const transferAmount = useMemo(() => {
```

Comment:
One transaction can include multiple coins (of different type) moved from and to an account. Maybe we should start showing that to the UI somehow instead of showing always 1 or nothing.

Reply:
done